lib/blob.js

```diff
@@ -0,0 +1,382 @@
+'use strict';
+
+var assert            = require('assert');
+var crypto            = require('crypto');
+var debug             = require('debug')('azure:blob');
+var Promise           = require('promise');
+var utils             = require('./utils');
+var querystring       = require('querystring');
+var xml               = require('./xml-parser');
+var Constants         = require('./constants');
+var HttpConstants     = Constants.HttpConstants;
+var HttpVerbs         = HttpConstants.HttpVerbs;
+var HttpResponseCodes = HttpConstants.HttpResponseCodes;
+
+/*
+ * Azure storage service version
+ * @const
+ */
+var SERVICE_VERSION = '2015-12-11';
+
+/* Transient error codes (we'll retry request when encountering these codes */
+var TRANSIENT_ERROR_CODES = [
+  // Azure error codes we should retry on according to azure docs
+  'InternalError',
+  'ServerBusy'
+].concat(utils.TRANSIENT_HTTP_ERROR_CODES);
+
+/*
+ * List of query-string parameter supported in lexicographical order, used for
+ * construction of the canonicalized resource.
+ */
+var QUERY_PARAMS_SUPPORTED = [
+  'comp',
+  'timeout',
+  'restype'
+].sort();
+
+/*
+ * Authorize the request with shared key
+ * Intended to define `Blob.prototype.authorize`.
+ */
+var authorizeWithSharedKey = function (method, path, query, headers) {
+  // Find account id
+  var accountId = this.options.accountId;
+
+  // Build string to sign
+  var stringToSign = (
+    method + '\n' +
+    (headers['content-encoding']     || '') + '\n' +
+    (headers['content-language']     || '') + '\n' +
+    (headers['content-length']       || '') + '\n' +
+    (headers['content-md5']          || '') + '\n' +
+    (headers['content-type']         || '') + '\n' +
+    '\n' + // we always include x-ms-date, so we never specify date header
+    (headers['if-modified-since']    || '') + '\n' +
+    (headers['if-match']             || '') + '\n' +
+    (headers['if-none-match']        || '') + '\n' +
+    (headers['if-unmodified-since']  || '') + '\n' +
+    (headers['range']                || '')
+  );
+
+  var fields;
+  // Construct fields as a sorted list of 'x-ms-' prefixed headers
+  fields = [];
```

Comment:
Do `var fields = []` in one line...

----
So generally speaking this seems like a solid start... My comments mostly concern really minor things :)